hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/shell/TestReplicationConfigPreference.java

```diff
@@ -0,0 +1,285 @@
+                                       String clientConfigReplType,
+                                       String clientConfigReplConfig,
+                                       String serverDefaultReplType,
+                                       String serverDefaultReplConfig) throws Exception {
+
+    for (String b: bucketList) {
+      OzoneBucket bucket = client.getObjectStore().getVolume(volumeName)
+          .getBucket(b);
+      if (b.equals(DEFAULT_BUCKET)) {
+        assertNull(bucket.getReplicationConfig());
+      } else {
+        assertNotNull(bucket.getReplicationConfig());
+      }
+
+      for (String k: keyList) {
+        OzoneKeyDetails key = bucket.getKey(k);
+        if (b.equals(DEFAULT_BUCKET) && k.equals(DEFAULT_KEY)) {
+          // if replication config are not passed as CLI params during bucket and
+          // key creation then key uses client configs.
+          // If replication is not set in client configs then key uses
+          // server default replication config.
+          if (clientConfigReplConfig != null) {
+            assertEquals(clientConfigReplType, key.getReplicationConfig()
+                .getReplicationType().name());
+            assertEquals(clientConfigReplConfig, key.getReplicationConfig()
+                .getReplication());
+          } else {
+            assertEquals(serverDefaultReplType, key.getReplicationConfig()
+                .getReplicationType().name());
+            assertEquals(serverDefaultReplConfig, key.getReplicationConfig()
+                .getReplication());
+          }
+        } else if (k.equals(DEFAULT_KEY)) {
+          // if replication config are not passed as CLI params
+          // during key creation and bucket replication config is set then
+          // key uses bucket replication config
+          // unless replication config is available in client configs.
+          if (clientConfigReplConfig != null) {
+            assertEquals(clientConfigReplType, key.getReplicationConfig()
+                .getReplicationType().name());
+            assertEquals(clientConfigReplConfig, key.getReplicationConfig()
+                .getReplication());
+          } else {
+            assertEquals(bucket.getReplicationConfig(),
+                key.getReplicationConfig());
+          }
+        } else if (k.equals(RATIS_KEY)) {
+          // if client sets replication config during key creation then
+          // this replication config is used regardless of client configs or
+          // bucket replication or server defaults.
+          assertEquals(RATIS_REPL_CONF, key.getReplicationConfig());
+        } else if (k.equals(EC_KEY)) {
+          assertEquals(EC_REPL_CONF, key.getReplicationConfig());
+        }
+      }
+    }
+  }
+
+  @Test
+  public void testReplicationOrderDefaultRatisCluster() throws Exception {
+    // Starting a cluster with server default replication type RATIS/THREE.
+    shutdown();
+    startCluster(new OzoneConfiguration());
```

Comment:
This is bad for test run time.  If the test cases require different clusters, then they should create their own, and `BeforeAll` does not need do so.  That requires N clusters (N=2 here), while with `BeforeAll` it is N+1.

Reply:
@adoroszlai perhaps I'm missing something but `BeforeAll` does not call `startCluster()`

Reply:
Oh, sorry, you are right.  I thought `shutdown` is called because it did, but it's just to make sure the test passes in any order.

Thanks.